{"version":1,"ops":[{"type":5,"author":{"id":"b83e8b4dfca7ebd9aba65d02ac0e5e24f95b7802"},"timestamp":1488899107,"metadata":{"github-id":"MDEyOkxhYmVsZWRFdmVudDk4OTcxMDU3NQ=="},"added":["bug"],"removed":[]},{"type":3,"author":{"id":"b83e8b4dfca7ebd9aba65d02ac0e5e24f95b7802"},"timestamp":1488924515,"metadata":{"github-id":"MDEyOklzc3VlQ29tbWVudDI4NDg3NTk4NQ==","github-url":"https://github.com/mindriot101/rust-fitsio/issues/25#issuecomment-284875985"},"message":"I haven't worked out why cfitsio is not raising an error here when trying to create a new image, but I've sorted out the segfault. It is because the test file ends with a table HDU, then when \"create_image\" tries to make an image but silently fails, my `fitsio` code assumes the current HDU has been switched to the new one, and then tries to read properties of the incorrect hdu type. I'm not really sure what's going on but cfitsio silently failing is the major source of this problem.","files":null},{"type":3,"author":{"id":"b83e8b4dfca7ebd9aba65d02ac0e5e24f95b7802"},"timestamp":1489008991,"metadata":{"github-id":"MDEyOklzc3VlQ29tbWVudDI4NTE3NzIxNg==","github-url":"https://github.com/mindriot101/rust-fitsio/issues/25#issuecomment-285177216"},"message":"It turns out that cfitsio only sets `status != 0` when closing the file, flushing the buffers to disk - hence why the error was not caught before segfaulting. The segfault was actually not related to writing to a readonly file at all, but in fact some other bug.\n\nI prefer to leave my \"hack\" in place, and check for file mutability before performing any mutable operations.","files":null},{"type":4,"author":{"id":"b83e8b4dfca7ebd9aba65d02ac0e5e24f95b7802"},"timestamp":1489008992,"metadata":{"github-id":"MDExOkNsb3NlZEV2ZW50OTkyMTI0OTQ2"},"status":2}]}